cdap-master/src/main/java/co/cask/cdap/data/tools/MDSUpgrader.java

```diff
@@ -235,15 +236,16 @@ private void handleRunRecordCompleted(final String appId, final String programId
     appMDS.executeUnchecked(new TransactionExecutor.Function<AppMDS, Void>() {
       @Override
       public Void apply(AppMDS appMetaStore) throws Exception {
-        List<RunRecord> completedRunRecords = appMetaStore.mds.list(partialKey, RunRecord.class);
-        for (RunRecord curCompletedRunRecord : completedRunRecords) {
+        Map<MDSKey, RunRecord> completedRunRecords = appMetaStore.mds.listKV(partialKey, RunRecord.class);
+        for (Map.Entry<MDSKey, RunRecord> curCompletedRunRecord : completedRunRecords.entrySet()) {
           // we cannot add a runRecordCompleted if there is no runRecordStarted for the run so write it first
           // the next setStop call deletes the runRecordStarted for that run so we don't have to delete it here.
-          writeTempRunRecordStart(appId, programType, programId, curCompletedRunRecord.getPid(),
-                                  curCompletedRunRecord.getStartTs());
```

Comment:
We don't put anything when we write stop record. The function is there to get start time for RunRecord of started.